bugsnag/client.py

```diff
@@ -192,16 +189,30 @@ def send_payload():
                     }
                 else:
                     event.severity_reason = initial_reason
+
                 payload = event._payload()
+
+                post_delivery_callback = self._request_tracker.new_request()
+                options = {'post_delivery_callback': post_delivery_callback}
+
+                if asynchronous is not None:
+                    options['asynchronous'] = asynchronous
+
                 try:
-                    self.configuration.delivery.deliver(self.configuration,
-                                                        payload, options)
+                    self.configuration.delivery.deliver(
+                        self.configuration,
+                        payload,
+                        options
+                    )
                 except Exception as e:
                     self.configuration.logger.exception(
                         'Notifying Bugsnag failed %s',
                         e
                     )
 
+                    # ensure this request is not still marked as in-flight
+                    post_delivery_callback()
```

Comment:
Given the standard delivery has finally statements to call this baked in, do we still need to have this here?  I guess in the case of an exception before we get to that point?

Reply:
Yeah, it's for exactly this:

> in the case of an exception before we get to that point?